### PR TITLE
Allow configuring ignored files when running check

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

File: src/towncrier/check.py [hunk 1]
```diff
@@ -52,27 +52,35 @@ def __main(comparewith, directory, config):
...
 
-    if not files_changed:
+    # Convert changed files to a list.
+    files_changed = files_changed.split(os.linesep) if files_changed else []
+
+    # Create a set of changed files converted to absolute paths.
+    files = set(os.path.join(base_directory, f) for f in files_changed)
+
+    # Filter out ignored files.
+    ignore_files = [os.path.join(base_directory, f) for f in config["check_ignore_files"]]
```
Comment (kostko) [inline src/towncrier/check.py]: Wouldn't it be better to generate a set here?

**Review by kostko — APPROVED** [review]
Can this be upstreamed?

File: src/towncrier/check.py (hunk 1 above)
Comment (tjanez) [reply src/towncrier/check.py]: Agreed. Done.

**Comment by tjanez** [comment]
> Can this be upstreamed?

Indeed, the change doesn't have anything that would be specific to us and I guess it could be useful to other users of towncrier.

However, the maintainers (even after the switch to the [Twisted Matrix Labs](https://github.com/twisted) organization for maintenance: https://github.com/twisted/towncrier/issues/159 in Apr 2020) aren't very active and there is a bunch of pull requests and issues pilling up currently...

Hence, I lost a bit of motivation to open upstream PRs...

File: src/towncrier/check.py [hunk 2]
```diff
@@ -52,27 +52,35 @@ def __main(comparewith, directory, config):
...
 
-    if files_changed == config["filename"]:
+    click.echo("Looking at these files:")
+    for n, f in enumerate(files, start=1):
+        click.echo("{}. {}".format(n, f))
+    click.echo()
+
+    if len(files) == 1 and files.pop() == os.path.join(base_directory, config["filename"]):
         click.echo("Only the configured news file has changed.")
         sys.exit(0)
```
Comment (lukaw3d) [inline src/towncrier/check.py:82]: https://github.com/oasisprotocol/towncrier/issues/11

`files.pop()` causes a bug: it removes a file even if it doesn't match `== os.path.join(...`.

https://github.com/oasisprotocol/rose-app/pull/234 fails:
- `files = ['/home/runner/work/rose-app/rose-app/.changelog/234.trivial.md']`
- `len(files) == 1` is true. continue to pop
- `files.pop() == CHANGELOG.md` is false
- `files` is now []
- `fragments_in_branch = fragments & files` is false :scream:

Comment (tjanez) [reply src/towncrier/check.py:82]: Nice find! @lukaw3d, can you prepare a fix for it?

Comment (lukaw3d) [reply src/towncrier/check.py:82]: This probably works https://github.com/oasisprotocol/towncrier/pull/12

**Pull request merged**